src/mds/Server.cc

```diff
@@ -5748,7 +5785,7 @@ int Server::parse_quota_vxattr(string name, string value, quota_info_t *quota)
           return r;
       }
     } else if (name == "quota.max_bytes") {
-      int64_t q = boost::lexical_cast<int64_t>(value);
+      int64_t q = return_bytes(value);
```

Comment:
The return value from `return_bytes()`  if negative causes the code to return `-CEPHFS_EINVAL`, which is good. But you might want to check the return value from `lexical_cast<>` and log the error about negative or zero input values and then return `-EINVAL`
e.g. `"-1G"` or `"0k"`

Reply:
`-1G` is quite understandable that it is not acceptable. If someone hits `-1G` or `0k`, isn't just `EINVAL` enough to hint to the user?

Reply:
since you embarked on logging the errors, I've mentioned this issue as a matter of completeness

Reply:
cool, maybe I can do `if(value <=0) {dout(10) << "non-postive value not permitted"}`

Reply:
Reminder!
Value [zero](https://docs.ceph.com/en/latest/cephfs/quota/#configuration) is valid and is used to discard any quotas set.

Reply:
yeah, i added a if block `if (value_str.front() == '-') {....}`, should be enough